src/transformers/models/pvt_v2/modeling_pvt_v2.py

```diff
@@ -0,0 +1,723 @@
+
+_CONFIG_FOR_DOC = "PvtV2Config"
+
+_CHECKPOINT_FOR_DOC = "FoamoftheSea/pvt_v2_b0"
+_EXPECTED_OUTPUT_SHAPE = [1, 256, 7, 7]
+
+_IMAGE_CLASS_CHECKPOINT = "FoamoftheSea/pvt_v2_b0"
+_IMAGE_CLASS_EXPECTED_OUTPUT = "LABEL_281"  # ImageNet ID for "tabby, tabby cat"
+
+PVT_V2_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "FoamoftheSea/pvt_v2_b0",
+    "FoamoftheSea/pvt_v2_b1",
+    "FoamoftheSea/pvt_v2_b2",
+    "FoamoftheSea/pvt_v2_b2_linear",
+    "FoamoftheSea/pvt_v2_b3",
+    "FoamoftheSea/pvt_v2_b4",
+    "FoamoftheSea/pvt_v2_b5",
+    # See all PVT models at https://huggingface.co/models?filter=pvt_v2
+]
+
+
+# Copied from transformers.models.beit.modeling_beit.drop_path
+def drop_path(input: torch.Tensor, drop_prob: float = 0.0, training: bool = False) -> torch.Tensor:
+    """
+    Drop paths (Stochastic Depth) per sample (when applied in main path of residual blocks).
+
+    Comment by Ross Wightman: This is the same as the DropConnect impl I created for EfficientNet, etc networks,
+    however, the original name is misleading as 'Drop Connect' is a different form of dropout in a separate paper...
+    See discussion: https://github.com/tensorflow/tpu/issues/494#issuecomment-532968956 ... I've opted for changing the
+    layer and argument names to 'drop path' rather than mix DropConnect as a layer name and use 'survival rate' as the
+    argument.
+    """
+    if drop_prob == 0.0 or not training:
+        return input
+    keep_prob = 1 - drop_prob
+    shape = (input.shape[0],) + (1,) * (input.ndim - 1)  # work with diff dim tensors, not just 2D ConvNets
+    random_tensor = keep_prob + torch.rand(shape, dtype=input.dtype, device=input.device)
+    random_tensor.floor_()  # binarize
+    output = input.div(keep_prob) * random_tensor
+    return output
+
+
+# Copied from transformers.models.convnext.modeling_convnext.ConvNextDropPath with ConvNext->Pvt
+class PvtV2DropPath(nn.Module):
+    """Drop paths (Stochastic Depth) per sample (when applied in main path of residual blocks)."""
+
+    def __init__(self, drop_prob: Optional[float] = None) -> None:
+        super().__init__()
+        self.drop_prob = drop_prob
+
+    def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
+        return drop_path(hidden_states, self.drop_prob, self.training)
+
+    def extra_repr(self) -> str:
+        return "p={}".format(self.drop_prob)
+
+
+class PvtV2OverlapPatchEmbeddings(nn.Module):
+    """Image to Patch Embedding"""
+
+    def __init__(
+        self,
+        config: PvtV2Config,
+        layer_idx: int,
+    ):
+        super().__init__()
+        patch_size: Union[int, Iterable[int]] = config.patch_sizes[layer_idx]
+        patch_size = (patch_size, patch_size) if isinstance(patch_size, int) else patch_size
+        stride: int = config.strides[layer_idx]
+        num_channels: int = config.num_channels if layer_idx == 0 else config.hidden_sizes[layer_idx - 1]
+        hidden_size: int = config.hidden_sizes[layer_idx]
```

Comment:
I'd prefer for the types here to be removed - the types of the arguments are pretty obvious from the code here 

```suggestion
        patch_size = config.patch_sizes[layer_idx]
        patch_size = (patch_size, patch_size) if isinstance(patch_size, int) else patch_size
        stride = config.strides[layer_idx]
        num_channels = config.num_channels if layer_idx == 0 else config.hidden_sizes[layer_idx - 1]
        hidden_size = config.hidden_sizes[layer_idx]
```